packages/flutter/lib/src/widgets/tween_animation_builder.dart

```diff
@@ -0,0 +1,249 @@
+  /// triggered at anytime by providing a new [Tween] with a new [Tween.end]
+  /// value. The new animation runs from the current animation value (which may
+  /// be [Tween.end] of the old [tween], if that animation completed) to
+  /// [Tween.end] of the new [tween]. The [Tween.begin] value is ignored except
+  /// for the initial animation that is triggered when the widget builds for the
+  /// first time.
+  ///
+  /// Any (subclass of) [Tween] is accepted as an argument. For example, to
+  /// animate the height or width of a [Widget], use a [Tween<double>], or
+  /// check out the [ColorTween] to animate the color property of a [Widget].
+  ///
+  /// Any [Tween] provided must have a non-null [Tween.end] value.
+  ///
+  /// ## Ownership
+  ///
+  /// The [TweenAnimationBuilder] takes full ownership of the provided [Tween]
+  /// and it will mutate the [Tween]. Once a [Tween] instance has been passed
+  /// to [TweenAnimationBuilder] its properties should not be accessed or
+  /// changed anymore to avoid any interference with the
+  /// [TweenAnimationBuilder]. If you need to change the [Tween], create a
+  /// **new instance** with the new values.
+  ///
+  /// It is good practice to never store a [Tween] provided to a
+  /// [TweenAnimationBuilder] in an instance variable to avoid accidental
+  /// modifications of the [Tween].
+  final Tween<T> tween;
+
+  /// Called every time the animation value changes.
+  ///
+  /// The current animation value is passed to the builder along with the
+  /// [child]. The builder should build a [Widget] based on the current
+  /// animation value and incorporate the [child] into it, if it is non-null.
+  final ValueWidgetBuilder<T> builder;
+
+  /// The child widget to pass to the builder.
+  ///
+  /// If a builder callback's return value contains a subtree that does not
+  /// depend on the animation, it's more efficient to build that subtree once
+  /// instead of rebuilding it on every animation tick.
+  ///
+  /// If the pre-built subtree is passed as the child parameter, the
+  /// [TweenAnimationBuilder] will pass it back to the [builder] function so
+  /// that it can be incorporated into the build.
+  ///
+  /// Using this pre-built child is entirely optional, but can improve
+  /// performance significantly in some cases and is therefore a good practice.
+  final Widget child;
+
+  /// Called every time an animation completes.
+  ///
+  /// This can be useful to trigger additional actions (e.g. another animation)
+  /// at the end of the current animation.
+  final VoidCallback onEnd;
+
+  @override
+  ImplicitlyAnimatedWidgetState<ImplicitlyAnimatedWidget> createState() {
+    return _TweenAnimationBuilderState<T>();
+  }
+}
+
+class _TweenAnimationBuilderState<T> extends AnimatedWidgetBaseState<TweenAnimationBuilder<T>> {
+  Tween<T> _currentTween;
+
+  @override
+  void initState() {
+    widget.tween.begin ??= widget.tween.end;
+    _currentTween = widget.tween;
```

Comment:
let's move line 209 to above line 208 and then use _currentTween when setting things. That way it'll be clearer that we're taking ownership.

Reply:
Done.